### PR TITLE
Vectorize RandomZoom

### LINKED ISSUE #1416: Vectorize RandomZoom
This PR should be base on:

https://github.com/keras-team/keras-cv/pull/1394

### PR DESCRIPTION
**Pull request by james77777778**
# What does this PR do?

Fixes #1416 

Benchmark results:

comparison

OldRandomZoom is so slow that I need to set `num_images = [100, 200, 500, 1000]` for benchmarks

Both fail to run with XLA due to lacking `ImageProjectiveTransformV3` support.
https://github.com/tensorflow/tensorflow/issues/55194

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/keras-team/keras-cv/blob/master/.github/CONTRIBUTING.md),
      Pull Request section?
- [X] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [X] Did you write any new necessary tests?
- [ ] If this adds a new model, can you run a few training steps on TPU in Colab to ensure that no XLA incompatible OP are used?

## Who can review?
@LukeWood @ianstenbit

### CODE REVIEW

File: keras_cv/layers/preprocessing/random_zoom.py [hunk 1]
```diff
@@ -28,7 +28,7 @@
 
 
 @tf.keras.utils.register_keras_serializable(package="keras_cv")
-class RandomZoom(BaseImageAugmentationLayer):
+class RandomZoom(VectorizedBaseImageAugmentationLayer):
```
Comment (LukeWood) [inline keras_cv/layers/preprocessing/random_zoom.py:31]: can we add a unit test to ensure that each image is zoomed independently?

This can be achieved by:

- generate an image using random.normal((1, 100, 100, 3))
- generate a batch with tf.stack([image, image], axis=0)
- call a random zoom layer
- assertNotAllClose(result[0], result[1])

This ensures we still have element-wise augmentation

**Review by LukeWood — CHANGES_REQUESTED** [review]
Thanks for the swift work on this PR.

I'd say that this looks good to me, just want one more unit test to be sure we don't have a regression on the element-wise operation.  The test in the benchmark does not actually cover this as we have a constant sample - so let's add an explicit unit test for it.

Thanks again for the hard work.

File: keras_cv/layers/preprocessing/random_zoom.py (hunk 1 above)
Comment (james77777778) [reply keras_cv/layers/preprocessing/random_zoom.py:31]: Done.

I added a unit test:
```python
    def test_random_zoom_on_batched_images_independently(self):
        image = tf.random.uniform(shape=(100, 100, 3))
        input_images = tf.stack([image, image], axis=0)

        layer = RandomZoom(
            height_factor=(-0.4, -0.5), width_factor=(-0.2, -0.3)
        )

        results = layer(input_images)
        self.assertNotAllClose(results[0], results[1])
```
and it passed as expected.

**Review by LukeWood — APPROVED** [review]
Awesome!  Thanks for the quick update.  This looks good to me!

**Comment by sebastian-sz** [comment]
@james77777778 good work with this vectorization.

I have one question:
This seems like geometrical transform, are functions `augment_bounding_boxes` and `augment_segmentation_masks` correct as passthrough?

I can see that they are not present in the non vectorized implementation.

**Comment by james77777778** [comment]
> @james77777778 good work with this vectorization.
> 
> I have one question: This seems like geometrical transform, are functions `augment_bounding_boxes` and `augment_segmentation_masks` correct as passthrough?
> 
> I can see that they are not present in the non vectorized implementation.

Yeah, they are just passthrough. I think the support for bbox and segmentation masks should be another PR?

**Comment by sebastian-sz** [comment]
@james77777778 
>Yeah, they are just passthrough. I think the support for bbox and segmentation masks should be another PR?

Maybe they could be left unimplemented? That way if someone uses them for Object Detection / Image Segmentation they will explicitly get an error. 
Otherwise, this might introduce silent error - if someone doesn't know that this is not yet implemented, they will use it and get incorrect boxes/masks.

**Comment by james77777778** [comment]
> @james77777778
> 
> > Yeah, they are just passthrough. I think the support for bbox and segmentation masks should be another PR?
> 
> Maybe they could be left unimplemented? That way if someone uses them for Object Detection / Image Segmentation they will explicitly get an error. Otherwise, this might introduce silent error - if someone doesn't know that this is not yet implemented, they will use it and get incorrect boxes/masks.

Thank you for pointing that out. I just copied these functions from another vectorized layer.
I'm going to remove `augment_bounding_boxes`, `augment_segmentation_masks` ASAP

**Comment by sebastian-sz** [comment]
@james77777778 Thanks! And great job with this layer!

**Comment by LukeWood** [comment]
> @james77777778
> 
> > Yeah, they are just passthrough. I think the support for bbox and segmentation masks should be another PR?
> 
> Maybe they could be left unimplemented? That way if someone uses them for Object Detection / Image Segmentation they will explicitly get an error. Otherwise, this might introduce silent error - if someone doesn't know that this is not yet implemented, they will use it and get incorrect boxes/masks.

ah good catch @sebastian-sz -- we should definitely be raising an error on these!  Thank you for the review.

File: keras_cv/layers/preprocessing/random_zoom.py [hunk 2]
```diff
@@ -138,60 +138,51 @@ def __init__(
...
             fill_value=self.fill_value,
             interpolation=self.interpolation,
         )
-        output = tf.squeeze(output, 0)
-        output.set_shape(original_shape)
-        return output
+        outputs.set_shape(original_shape)
+        return outputs
 
-    def augment_label(self, label, transformation, **kwargs):
```
Comment (LukeWood) [inline keras_cv/layers/preprocessing/random_zoom.py:181]: label can be a passthrough

**Review by LukeWood — APPROVED** [review]

**Comment by LukeWood** [comment]
Thank for the great contribution @james77777778 !!!
